game/engine.py

```
@@ -367,15 +733,70 @@ def is_promotion_move(board, fr, fc, tr):
+                    notation = f"K{t_coord}"
+        else:
+            # Fallback: simplified notation
+            files = "abcdefgh"
+            f_coord = f"{files[fc]}{8 - fr}"
+            t_coord = f"{files[tc]}{8 - tr}"
+
+            if not piece:
+                notation = f"{f_coord} -> {t_coord}"
+
+            else:
+                type = piece.lower()
```

Ruff (0.15.13)

[error] Variable `type` is shadowing a Python builtin

(A001)